generators/new_collateral/config/parser.go

```diff
@@ -0,0 +1,85 @@
+package config
+
+import (
+	"errors"
+
+	"github.com/BurntSushi/toml"
+	"github.com/makerdao/vdb-mcd-transformers/generators/new_collateral/helpers"
+	"github.com/makerdao/vdb-mcd-transformers/generators/new_collateral/types"
+	"github.com/mitchellh/mapstructure"
+)
+
+type IParse interface {
+	ParseCurrentConfig(configFilePath, configFileName string) (types.TransformersConfig, error)
+}
+type Parser struct{}
+
+func (Parser) ParseCurrentConfig(configFilePath, configFileName string) (types.TransformersConfig, error) {
+	var tomlConfig types.TransformersConfigForToml
+	fullConfigFilePath := helpers.GetFullConfigFilePath(configFilePath, configFileName)
+	_, decodeErr := toml.DecodeFile(fullConfigFilePath, &tomlConfig)
+	if decodeErr != nil {
+		return types.TransformersConfig{}, decodeErr
+	}
+
+	metadata, metadataErr := parseExporterMetaData(tomlConfig)
+	if metadataErr != nil {
+		return types.TransformersConfig{}, metadataErr
+	}
+
+	transformerExporters, transformerExportersErr := parseTransformerExporters(tomlConfig)
+	if transformerExportersErr != nil {
+		return types.TransformersConfig{}, transformerExportersErr
+	}
+
+	return types.TransformersConfig{
+		ExporterMetadata:     metadata,
+		TransformerExporters: transformerExporters,
+		Contracts:            tomlConfig.Contracts,
+	}, nil
+}
+
+func parseExporterMetaData(tomlConfig types.TransformersConfigForToml) (types.ExporterMetaData, error) {
+	home, homeOk := tomlConfig.Exporter["home"].(string)
+	name, nameOk := tomlConfig.Exporter["name"].(string)
+	save, saveOk := tomlConfig.Exporter["save"].(bool)
+	schema, schemaOk := tomlConfig.Exporter["schema"].(string)
+	if !homeOk || !nameOk || !saveOk || !schemaOk {
+		return types.ExporterMetaData{}, errors.New("error asserting exporter meta data types")
+	}
+
+	var transformerNames []string
+	decodeErr := mapstructure.Decode(tomlConfig.Exporter["transformerNames"], &transformerNames)
```

Comment:
do we want to check `tomlConfig.Exporter["transformerNames"]` with a similar `ok` pattern? Not sure if we could run into null pointer errors or silent failures here

Reply:
I just tested it out, and it looks like if `tomlConfig.Exporter["transformerNames"]` were nil, `mapstructure` would just decode it into an empty `[]string`, which I think is ok. Happy to reconsider this, and add a check since we probably don't want an empty slice of transformer names, but OTHO it shouldn't break anything.

Reply:
wrote a test for it in 938de7a859b2a550b95a0efe656128c314b15f6e